### PR TITLE
KATI-40_Condition/Symptom_Info

### PR DESCRIPTION
**Pull request by toannguyen71195**

### CODE REVIEW

**Comment by hieumai** [comment]
@toannguyen71195  please resolve conflicts first, then i will review

File: app/src/main/java/com/kms/cura/view/activity/SymptomInfoActivity.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+import com.kms.cura.entity.SymptomEntity;
+import com.kms.cura.view.fragment.AddDialogFragment;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SymptomInfoActivity extends AppCompatActivity implements View.OnClickListener, AdapterView.OnItemClickListener, Toolbar.OnMenuItemClickListener {
+
+    public static final String DIALOG_TAG = "addDialog";
+    public static final String ADD_TITLE = "Details about your symptom";
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/activity/SymptomInfoActivity.java]: This should be in resource file

File: app/src/main/java/com/kms/cura/view/fragment/AddDialogFragment.java [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        if (year > calendar.get(Calendar.YEAR)) {
+            return false;
+        }
+        if (month > calendar.get(Calendar.MONTH) + 1) {
+            return false;
+        }
+        if (day > calendar.get(Calendar.DAY_OF_MONTH)) {
+            return false;
+        }
+        return true;
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/fragment/AddDialogFragment.java]: Should use java Calendar before or after for this method

File: app/src/main/java/com/kms/cura/view/fragment/AddDialogFragment.java [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+        if (year < startYear) {
+            return false;
+        }
+        if (month < startMonth) {
+            return false;
+        }
+        if (day < startDay) {
+            return false;
+        }
+        return true;
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/fragment/AddDialogFragment.java]: same for this method

File: com.kms.cura.dal/src/com/kms/cura/dal/database/ConditionDatabaseHelper.java [hunk 4]
```diff
@@ -1,24 +1,63 @@
...
+	}
+
+	public List<ConditionEntity> queryAssociatedCondition(SymptomEntity symptomEntity)
+			throws SQLException, ClassNotFoundException {
+		PreparedStatement stmt = null;
+		ResultSet rs = null;
+		StringBuilder builder = new StringBuilder();
+		List<ConditionEntity> conditionEntities = new ArrayList<>();
+		// SELECT * FROM cura.conditions_symptoms s left join cura.conditions c
+		// on s.condition_id = c.id where symptom_id = ?
```
Comment (hieumai) [inline com.kms.cura.dal/src/com/kms/cura/dal/database/ConditionDatabaseHelper.java]: Remove unused comments

File: com.kms.cura.dal/src/com/kms/cura/dal/database/SpecialityDatabaseHelper.java [hunk 5]
```diff
@@ -20,8 +22,38 @@ public SpecialityDatabaseHelper() throws ClassNotFoundException, SQLException {
...
+	protected SpecialityEntity getEntityFromResultSet(ResultSet resultSet) throws SQLException, ClassNotFoundException {
 		return new SpecialityEntity(resultSet.getString(DegreeColumn.ID.getColumnName()), resultSet.getString(DegreeColumn.NAME.getColumnName()));
 	}
 	
+	public List<SpecialityEntity> querySpecialityByCondition(ConditionEntity entity) throws SQLException, ClassNotFoundException {
+		PreparedStatement stmt = null;
+		ResultSet rs = null;
+		StringBuilder builder = new StringBuilder();
+		List<SpecialityEntity> specialityEntities = new ArrayList<>();
+		// SELECT * FROM cura.specialities_conditions c left join cura.specialties s on c.speciality_id = s.id where condition_id = 2;
```
Comment (hieumai) [inline com.kms.cura.dal/src/com/kms/cura/dal/database/SpecialityDatabaseHelper.java]: Remove comment here too

File: com.kms.cura.dal/src/com/kms/cura/dal/database/SymptomDatabaseHelper.java [hunk 6]
```diff
@@ -13,8 +19,42 @@ public SymptomDatabaseHelper() throws ClassNotFoundException, SQLException {
...
 	}
+
+	public List<SymptomEntity> queryAssociatedSymptom(ConditionEntity conditionEntity)
+			throws SQLException, ClassNotFoundException {
+		PreparedStatement stmt = null;
+		ResultSet rs = null;
+		StringBuilder builder = new StringBuilder();
+		List<SymptomEntity> symptomEntities = new ArrayList<>();
+		// SELECT * FROM cura.conditions_symptoms c left join cura.symptoms s on
+		// c.symptom_id = s.id where condition_id = ?
```
Comment (hieumai) [inline com.kms.cura.dal/src/com/kms/cura/dal/database/SymptomDatabaseHelper.java]: And here

File: com.kms.cura_server/src/com/kms/cura_server/ConditionAPI.java [hunk 7]
```diff
@@ -24,4 +30,17 @@ public String getAllCondition() {
 			return APIResponse.unsuccessResponse(e.getMessage());
 		}
 	}
+	
+	@POST
+	@Path("/getAssociatedCondition")
+	public String getAssociatedSymptom(String jsonData) {
```
Comment (hieumai) [inline com.kms.cura_server/src/com/kms/cura_server/ConditionAPI.java]: Wrong method name. Should be getAssociatedCondition

File: com.kms.cura_server/src/com/kms/cura_server/SpecialityAPI.java [hunk 8]
```diff
@@ -33,6 +39,16 @@ public String getAllDegree() {
 		}
 	}
 	
-
-	
+	@POST
+	@Path("/getByCondition")
+	public String getByCondtion(String jsonData) {
```
Comment (hieumai) [inline com.kms.cura_server/src/com/kms/cura_server/SpecialityAPI.java]: Typo on getByCondtion

**Comment by toannguyen71195** [comment]
@hieumai Fixed

**Comment by hieumai** [comment]
OK from me. @vtnlinh95  and @huuduy216  please start verifying on this

**Comment by vtnlinh95** [comment]
@toannguyen71195  : 
- Don't know why but everytime I click on one condition or symptom, a dialog appears with "null" message.
- It can not load the associate conditions and symptom
- When clicking back, it stopped working
- When navigating to doctor search, the specialities are not filled in
- Should show a dialog instead of toast when the users choose the wrong start date or end date

**Comment by toannguyen71195** [comment]
@vtnlinh95 fixed

**Comment by hieumai** [comment]
@toannguyen71195  please rebase this so I can merge

**Pull request merged**